FirebaseInstallations/Source/Library/FIRInstallations.m

```diff
@@ -143,6 +143,40 @@ + (void)validateAppOptions:(FIROptions *)appOptions appName:(NSString *)appName
             kFIRLoggerInstallations, kFIRInstallationsMessageCodeInvalidFirebaseAppOptions,
             [missingFields componentsJoinedByString:@", "]];
   }
+
+  [self validateAPIKey:appOptions.APIKey];
+}
+
++ (void)validateAPIKey:(nullable NSString *)APIKey {
+  NSMutableArray<NSString *> *validationIssues = [[NSMutableArray alloc] init];
+
+  if (APIKey.length != kExpectedAPIKeyLength) {
+    [validationIssues addObject:[NSString stringWithFormat:@"API Key length must be %lu characters",
+                                                           (unsigned long)kExpectedAPIKeyLength]];
+  }
+
+  if (![[APIKey substringToIndex:1] isEqualToString:@"A"]) {
+    [validationIssues addObject:@"API Key must start with `A`"];
+  }
+
+  NSCharacterSet *characters = [NSCharacterSet characterSetWithCharactersInString:APIKey];
+  if (![[NSCharacterSet alphanumericCharacterSet] isSupersetOfSet:characters]) {
+    [validationIssues addObject:@"API Key must contain only alphanumeric characters"];
+  }
```

Comment:
Looking at this code, I understand your concern. Mainly, because I don't know what is happening behind the scense.
If you cannot use regex, I would be fine if you check first character and length, and leave a comment, that only base64 url-safe characters are allowed `[\w-]`.